Erupe/server/channelserver/handlers.go

```diff
@@ -356,9 +357,100 @@ func handleMsgMhfAcquireTitle(s *Session, p mhfpacket.MHFPacket) {}
 
 func handleMsgMhfEnumerateTitle(s *Session, p mhfpacket.MHFPacket) {}
 
-func handleMsgMhfEnumerateUnionItem(s *Session, p mhfpacket.MHFPacket) {}
+type Item struct {
+	ItemId uint16
+	Amount uint16
+}
 
-func handleMsgMhfUpdateUnionItem(s *Session, p mhfpacket.MHFPacket) {}
+func handleMsgMhfEnumerateUnionItem(s *Session, p mhfpacket.MHFPacket) {
+	pkt := p.(*mhfpacket.MsgMhfEnumerateUnionItem)
+	var boxContents []byte
+	bf := byteframe.NewByteFrame()
+	err := s.server.db.QueryRow("SELECT item_box FROM users, characters WHERE characters.id = $1 AND users.id = characters.user_id", int(s.charID)).Scan(&boxContents)
+	if err != nil {
+		s.logger.Fatal("Failed to get shared item box contents from db", zap.Error(err))
+	} else {
+		if len(boxContents) == 0 {
+			bf.WriteUint32(0x00)
+		} else {
+			amount := len(boxContents) / 4
+			bf.WriteUint16(uint16(amount))
+			bf.WriteUint32(0x00)
+			bf.WriteUint16(0x00)
+			for i := 0; i < amount; i++ {
+				bf.WriteUint32(binary.BigEndian.Uint32(boxContents[i*4 : i*4+4]))
+				if i+1 != amount {
+					bf.WriteUint64(0x00)
+				}
+			}
+		}
+	}
+	doAckBufSucceed(s, pkt.AckHandle, bf.Data())
+	doAckBufSucceed(s, pkt.AckHandle, bf.Data())
```

Comment:
Here you are sending the buffered data twice. I don't think this causes any issues but it should be cut.

Reply:
Good catch ill remove.